frontend/src/concepts/modelRegistry/utils.ts

```diff
@@ -113,3 +114,80 @@ export const filterArchiveModels = (registeredModels: RegisteredModel[]): Regist
 
 export const filterLiveModels = (registeredModels: RegisteredModel[]): RegisteredModel[] =>
   registeredModels.filter((rm) => rm.state === ModelState.LIVE);
+
+/**
+ * Converts model source properties to catalog parameters
+ * @param modelSourceKind - The kind of model source
+ * @param modelSourceClass - The class of model source (maps to catalog source name)
+ * @param modelSourceGroup - The group of model source (maps to catalog repository name)
+ * @param modelSourceName - The name of model source (maps to catalog model name)
+ * @param modelSourceId - The ID of model source (maps to catalog model tag)
+ * @returns CatalogModelDetailsParams object or null if not a catalog source
+ */
+export const modelSourcePropertiesToCatalogParams = (
+  modelSourceKind?: string,
+  modelSourceClass?: string,
+  modelSourceGroup?: string,
+  modelSourceName?: string,
+  modelSourceId?: string,
+): CatalogModelDetailsParams | null => {
+  if (modelSourceKind !== ModelSourceKind.CATALOG) {
+    return null;
+  }
+
+  return {
+    sourceName: modelSourceClass || '',
+    repositoryName: modelSourceGroup || 'default',
+    modelName: modelSourceName || '',
+    tag: modelSourceId || '',
+  };
+};
```

Comment:
If you require the properties you need to be defined up front you don't need the fallbacks to `''` (also `'default'` isn't a valid repository name).

```ts
if (
  modelSourceKind !== ModelSourceKind.CATALOG ||
  !modelSourceClass ||
  !modelSourceGroup ||
  !modelSourceName ||
  !modelSourceId
) {
  return null;
}
```

that'll narrow the types to defined when you return them.